### PR TITLE
feat: Support ScalaPB generated envoy validation #695

### PR DESCRIPTION
**Pull request by johanandren**
References
- #695

### CODE REVIEW

**Comment by johanandren** [comment]
Not sure if we should make it more official and add to docs or if this is enough for now. Setting it up was kinda tricky to get right so that could be an argument for showing it off in docs.

**Comment by johanandren** [comment]
Curiously we generate all the message classes twice, once for main/prod and once for test (noticed when tests classes weren't validated), seems like something we should look into at some point.

Couldn't think of a reasonable way to trigger the validation failure from a test since the pb generated classes don't allow invalid message creation.

File: samples/scala-protobuf-validated/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+```shell
+curl -XPOST -H "Content-Type: application/json" localhost:9000/com.example.validated.Validated/CallMeMaybe -d '{"email":"notanemail"}'
+```
+
+Which results in a `403 BadRequest` HTTP response with a text body with details about what was wrong
+
+Or a proper email:
+```shell
+curl -XPOST -H "Content-Type: application/json" localhost:9000/com.example.validated.Validated/CallMeMaybe -d '{"email":"notanemail"}'
```
Comment (efgpinto) [inline samples/scala-protobuf-validated/README.md:39]:
```suggestion
curl -XPOST -H "Content-Type: application/json" localhost:9000/com.example.validated.Validated/CallMeMaybe -d '{"email":"very.email@example.com"}'
```

File: samples/scala-protobuf-validated/src/test/scala/com/example/validated/ValidatedActionIntegrationSpec.scala [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  private val client = testKit.getGrpcClient(classOf[Validated])
+
+  "ValidatedAction" must {
+
+    "handle command CallMeMaybe" in {
+      val result = client.callMeMaybe(Request("valid.email@example.com")).futureValue
+      result should ===(Response("Hello valid.email@example.com, gosh what a valid email you sent me"))
+
+      // we can't test validation failure because we can't construct an invalid request
```
Comment (efgpinto) [inline samples/scala-protobuf-validated/src/test/scala/com/example/validated/ValidatedActionIntegrationSpec.scala:31]: Did you consider enabling http transcoding and then having a http client here to make the invalid request? But then we would need to hardcode the host I guess.. 🤔 Probably not worth the hassle.

**Review by efgpinto — APPROVED** [review]
LGTM with suggested typo fix.

Had a note about test the failure but maybe not worth the trouble.

File: samples/scala-protobuf-validated/src/test/scala/com/example/validated/ValidatedActionIntegrationSpec.scala (hunk 2 above)
Comment (johanandren) [reply samples/scala-protobuf-validated/src/test/scala/com/example/validated/ValidatedActionIntegrationSpec.scala:31]: Yeah, would need to figure out the test container port, which seemed to tricky to be worth it.

**Pull request merged**